modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/JournalTransformer.java

```diff
@@ -344,6 +361,8 @@ else if (exception instanceof TransformException) {
 				httpServletRequest.setAttribute(
 					PortletRequest.LIFECYCLE_PHASE, originalLifecyclePhase);
 			}
+
+			renderedArticleIds.remove(articleId);
```

Comment:
@yuhai As a post-execution clean-up, this is not guaranteed to run. The try-finally scope is pretty small compared to the length of this method. Please consider this.